### ISSUE TITLE
Question: UILabels with Dynamic Height

### ISSUE THREAD

**Issue by SpacyRicochet** [Mubaloo/wwdc2015-uistackviews#1]
Have you had any luck with implementing a UIStackView with a label in it with a dynamic height? I'm trying to reimplement UIAlertController as a UIStackView exercise. This is tripping me up.

![screen shot 2015-06-30 at 16 14 12](https://cloud.githubusercontent.com/assets/904596/8432938/4f4679e8-1f43-11e5-933e-89081a5ca16a.png)

**Comment by rhodgkins** [comment]
Yeah - I've come across a similar issue. I'm not sure if the `UILabel` is ignoring its `preferredMaxLayoutWidth` or its something else. Hopefully it'll be fixed by final release.

I haven't tried this, but have you looked at sub-classing `UILabel` and then in `layoutSubviews` setting `preferredMaxLayoutWidth` to `bounds.width - 1`?

Cheers,

Rich

**Comment by JonathanDowning** [comment]
I've also experienced this issue (as of iOS 9/Xcode 7 beta 3) and doing what rhodgkins suggested worked!

``` swift
class FixedLabel: UILabel {

    override func layoutSubviews() {
        super.layoutSubviews()
        preferredMaxLayoutWidth = bounds.width - 1
    }

}
```

I also hope that this is fixed before final release :)

**Comment by aehlke** [comment]
Anyone found alternative solutions? The last one pegs CPU (infinite loop of `layoutSubviews` re-triggering)

**Comment by liamnichols** [comment]
What exactly are you doing with them for it to kill the CPU? It all seems to kinda just work in iOS 9 from what I've seen
